File: gui_apps/PyDoodle/step01/step01.py
"""
// ============================================================================
// step01.py: creating the basic Doodle PyQt application
//
// Tutorial - PyQt5 Doodle Application
// Based on a similar tutorial for Borland ObjectWindows Library (OWL)
// @author: Manish Bhobe
// My experiments with the Qt Framework. Use at your own risk!!
// ============================================================================
"""
import sys
import os
import pathlib
from PyQt5.QtGui import *
from mainWindow import *
import darkdetect

sys.path.append(os.path.join(pathlib.Path(__file__).parents[1], 'common'))
from mypyqt5_utils import PaletteSwitcher, PyQtApp


def main():
    app = PyQtApp(sys.argv)

    mainWindow = MainWindow()
    mainWindow.show()

    return app.exec_()


if __name__ == "__main__":
    main()

File: bogo2bogo/ChocolafStyle/examples/C++/MyProjects/Doodle/step01/step01.cc
// ============================================================================
// step01.cc: Creating a basic application with Qt/C++
//
// Tutorial - Qt Scribble Application
// Based on a similar tutorial for Borland ObjectWindows Library (OWL)
//
// @author Manish Bhobe for Nämostuté Ltd.
// My experiments with C++,Qt, Python & PyQt.
// Code is provided for illustration purposes only! Use at your own risk.
// =============================================================================
#include "chocolaf.h"
#include <QApplication>
#include <QMainWindow>
#include <QtGui>

int main(int argc, char **argv)
{
  QApplication app(argc, argv);
  // use Chocolaf style
  QFile f(":chocolaf/chocolaf.css");
  if (!f.exists()) {
    printf("Unable to open stylesheet!");
  } else {
    f.open(QFile::ReadOnly | QFile::Text);
    QTextStream ts(&f);
    app.setStyleSheet(ts.readAll());
  }

  // create the GUI
  QMainWindow mainWindow;
  QString title =
      QString("Qt %1 Doodle with Chocolaf - Step01: Basic Window").arg(QT_VERSION_STR);
  mainWindow.setWindowTitle(title);
  mainWindow.resize(QGuiApplication::primaryScreen()->availableSize() * 4 / 5);
  mainWindow.show();

  return app.exec();
}
